tests/queries/0_stateless/01744_alter_replace_partition_update.sql

```diff
@@ -0,0 +1,66 @@
+drop table if exists x;
+
+create table x (type Enum8('hourly' = 0, 'hourly_staging' = 1, 'daily' = 2, 'daily_staging' = 3), dt DateTime, i int)
+engine MergeTree partition by (type, if(toUInt8(type) < 2, dt, toStartOfDay(dt))) order by i;
+
+insert into x values ('daily', '2021-03-28 01:00:00', 10), ('daily', '2021-03-28 02:00:00', 20);
+insert into x values ('hourly', '2021-03-28 05:00:00', 60), ('hourly', '2021-03-28 06:00:00', 70);
+insert into x values ('hourly', '2021-03-28 07:00:00', 80), ('hourly', '2021-03-28 08:00:00', 90);
+insert into x values ('hourly', '2021-03-29 05:00:00', 60), ('hourly', '2021-03-29 06:00:00', 70);
+
+select * from x order by type, dt, i;
+
+-- replace from existing partition to existing partition
+alter table x replace partition ('daily', '2021-03-28 00:00:00') update type = 'daily' in partition ('hourly', '2021-03-28 07:00:00');
```

Comment:
Is it supposed to leave the initial partition inside `in partition` clause unmodified? If so, the I suggest to rename the clause to something like `from partition`, otherwise it's misleading.

Reply:
I expect the following here:

- `('daily', '2021-03-28 00:00:00')` partition gets removed (2 rows)
- `('hourly', '2021-03-28 07:00:00')` partition gets updated and renamed (1 row)

Reply:
`replace partition <dst_partition> update ... in partition <src_partition>`

replace `dst_partition` with updates from `src_partition`. `src_partition` remains unmodified.

Reply:
> `src_partition` remains unmodified.

Then I suggest to change `IN PARTITION` to `FROM PARTITION` - just for clarity. AFAIK there is no similar syntax in other clauses so it should be totaly fine.

Reply:
Ok. I'll change to `ALTER TABLE <name> REPLACE PARTITION <partition_expr> FROM PARTITION <partition_expr> UPDATE <update_expr>`